woocommerce/woocommerce
# Update/38743 loading changes in core profiler

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Some minor UI fixes in this PR:
- Spinners no longer appear between pages in the core profiler.
- Spinner was added to the 'Continue' button in the Business Info page, to indicate that it takes a moment to finish its' API calls
- Extensions screen is skipped when all available plugins are installed
- Loading screen after Extensions screen should have a 3rd loader step
- Skipping setup has a new customised loader

Closes #38743  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate WooCommerce on a brand new site
1. Install the WooCommerce Beta Tester plugin
1. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper
1. Enable the 'core-profiler' feature flag
1. Visit your public URL /wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
1. Go through the new core profiler flow.
1. Observe that spinners no longer appear between pages
1. Observe that upon submission of the Business Info page, the button shows a spinner until the page changes
1. Observe that there is a 3rd loader step in the loader during extensions installation
1. After a first runthrough installing all the plugins, a second runthrough should not show the extensions page
1. Clicking on "Skip guided setup" on the first page and going through the skip sequence should show a customised loader now

Extensions installation 3rd loader step:
![image](https://github.com/woocommerce/woocommerce/assets/27843274/07e57b44-7c20-4022-848b-fb6db40e5d8f)

Skip Guided Setup loader:
![image](https://github.com/woocommerce/woocommerce/assets/27843274/c1743649-6adc-4b16-be04-73aa43332859)
